packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/TerariumAssetServices.java

```diff
@@ -0,0 +1,40 @@
+package software.uncharted.terarium.hmiserver.service.data;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import software.uncharted.terarium.hmiserver.models.TerariumAsset;
+import software.uncharted.terarium.hmiserver.models.dataservice.AssetType;
+
+@Service
+@RequiredArgsConstructor
+public class TerariumAssetServices {
+
+	private final ArtifactService artifactService;
+	private final CodeService codeService;
+	private final DatasetService datasetService;
+	private final DocumentAssetService documentAssetService;
+	private final ModelConfigurationService modelConfigurationService;
+	private final ModelService modelService;
+	private final NotebookSessionService notebookSessionService;
+	private final WorkflowService workflowService;
+
+	/**
+	 * Get the service for a given asset type
+	 * @param type The type of asset to get the service for
+	 * @return The service for the given asset type
+	 */
+	public TerariumAssetService<? extends TerariumAsset> getServiceByType(final AssetType type) {
+        return switch (type) {
+            case ARTIFACT -> artifactService;
```

Comment:
I have nothing but good things to say about this new syntax for `switch` statements :)